webpack.config.js

```diff
@@ -379,6 +380,112 @@ var config = {
     ],
 };
 
+async function initializeModuleFederation() {
+    function makeSingletonSharedModules(packageNames) {
+        const sharedObject = {};
+
+        for (const packageName of packageNames) {
+            const version = packageJson.dependencies[packageName];
+
+            sharedObject[packageName] = {
+                requiredVersion: version,
+                singleton: true,
+                version,
+            };
+        }
+
+        return sharedObject;
+    }
+
+    function isWebpackDevServerAvailable(baseUrl) {
+        return new Promise((resolve) => {
+            if (!DEV) {
+                resolve(false);
+                return;
+            }
+
+            const req = http.request(`${baseUrl}/remote_entry.js`, (response) => {
+                return resolve(response.statusCode === 200);
+            });
+
+            req.on('error', () => {
+                resolve(false);
+            });
+
+            req.end();
+        });
+    }
+
+    async function getRemoteModules() {
+        const products = [
+            {name: 'focalboard', baseUrl: 'http://localhost:9006'},
+        ];
+
+        const productsFound = await Promise.all(products.map((product) => isWebpackDevServerAvailable(product.baseUrl)));
+
+        const remotes = {};
+        const aliases = {};
+
+        for (let i = 0; i < products.length; i++) {
+            const product = products[i];
+            const found = productsFound[i];
+
+            if (found) {
+                console.log(`Product ${product.name} found, adding as remote module`);
+
+                remotes[product.name] = `${product.name}@${product.baseUrl}/remote_entry.js`;
```

Comment:
If the product's dev server is running, we have to give the web app's Webpack the path to it. The `remote_entry.js` name isn't particularly significant, but it has to match the `filename` specified [here](https://github.com/mattermost/focalboard/pull/3748/files#diff-1c5589d2b7076ec8b90835ff95b065e6f57d7330758e104e1f853e079368c7c2R149)